README.md

```diff
@@ -291,6 +291,57 @@ writeResultFile("hostFunction.result", hostFunctionResult);
 ```
 -->
 
+### Store and Instantiating Multiple Modules
+
+A [Store][spec] is an intermediate-level abstraction that collects Wasm function, global, memory, and table instances
+as named entities.
+
+It simplifies creating instances when there are a lot of interdependencies, by collecting all the
+
+In the simplest case, it allows to register single host functions, globals, memories and tables:
+
+```java
+import com.dylibso.chicory.runtime.Store;
+
+// instantiate the store
+var store = new Store();
+// registers `console.log` defined above in the store
+store.addHostFunction(func);
+```
+
+However, the store also automatically exposes the exports of a module to the other instances that are registered.
+
+```java
+// registers `logger` into the store
+store.register("logger", instance);
+// now the exported function `logIt` can be imported by other modules as `logger.logIt`
+```
+
+There is also a shorthand method to instantiate a module and register the resulting instance:
+
+```java
+var logger2 = store.instantiate("logger2", Parser.parse(new File("./logger.wasm")));
+```
+
+This is equivalent to:
+
+```java
+var hostImports = store.toHostImports();
```

Comment:
I'm surprised that we are keeping compatibility with the "legacy" `HostImports`.
I think that using the `Store` covers already all the use cases, so, we can drop the previous lower level API.
wdyt?

Reply:
`HostImports` is just a bare container, and it is immutable. I felt it made sense to keep the distinction for now, we can always refine later :)

Reply:
ok, let's follow up on this.